pkg/service/signal.go

```diff
@@ -73,6 +73,19 @@ func NewDefaultSignalServer(
 		prometheus.IncrementParticipantRtcInit(1)
 
 		if rr, ok := router.(*routing.RedisRouter); ok {
+			rtcNode, err := router.GetNodeForRoom(ctx, roomName)
+			if err != nil {
+				return err
+			}
+
+			if rtcNode.Id != currentNode.Id {
+				err = routing.ErrIncorrectRTCNode
```

Comment:
Trying to understand this race. How does this happen?

Reply:
room node assignment isn't atomic... i'm not sure how since `StandardRoomAllocator.CreateRoom` acquires a lock before persisting the assignment but if more than one participants joins at the same time they can be assigned different nodes. someone reported this in slack and this is the last bit of legacy behavior that isn't present in signal relay - the solution is borrowed from https://github.com/livekit/livekit/blob/master/pkg/routing/redisrouter.go#L225-L237